src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -85,6 +85,8 @@ def _determine_rgw_addr(daemon_info: Dict[str, Any]) -> RgwDaemon:
     daemon = RgwDaemon()
     daemon.host = _parse_addr(daemon_info['addr'])
```

Comment:
One-liner, so the `daemon.host` is only set once:
```suggestion
    # Use hostname for SSL, IP address otherwise
    daemon.host = daemon_info['metadata']['hostname'] if daemon.ssl else _parse_addr(daemon_info['addr'])
```
That said, if we move to hostname-based connection, why not doing it also for HTTP connections?
```suggestion
    daemon.host = daemon_info['metadata']['hostname']
```

Reply:
>  That said, if we move to hostname-based connection, why not doing it also for HTTP connections?

good point, I added your suggestion to the code